### ISSUE TITLE
Jaeger grpc receiver ignores `span.Process` attribute and only accounts for `batch.Process` when mapping to `Resource`

### ISSUE THREAD

**Issue by oxeye-yuval** [open-telemetry/opentelemetry-collector-contrib#10594]
**Describe the bug**
When using the `jaeger-grpc-reciever` any `batch` received has it's `serviceName` attribute extracted from the `batch.Process` attribute, this is problematic as the `jaeger-idl` specifies that the `serviceName` field inside `span.Process` should have priority.
Looking at the code of the `jaeger-grpc-reciever` it looks like the `span.Process` field is ignored.

The jaeger receiver calls:
https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/9b8da699721b3360dc4cc2e5a7e74d74630ced40/receiver/jaegerreceiver/trace_receiver.go#L284
Which in turn calls to:
https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/9b8da699721b3360dc4cc2e5a7e74d74630ced40/pkg/translator/jaeger/jaegerproto_to_traces.go#L60
That maps the `batch.Process` field into a resource rather than `span.Process`

The jaeger-idl:
https://github.com/jaegertracing/jaeger-idl/blob/05fe64e9c305526901f70ff692030b388787e388/proto/api_v2/model.proto#L152-L160

The specification:
https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/sdk_exporters/jaeger.md#resource

**Steps to reproduce**
I used an instrumented `python-flask` server configured with a `jaeger-grpc-exporter` which sets the `span.Process` for the exported Spans (and not `batch.Process`).
A `jaeger-receiver` on the collector side with `logging-exporter` which shows that no resources are received (while they are actually configured for the exported span).

**What did you expect to see?**
I expected to see that the `span.Process`  resources get mapped to the `ptrace.ResourceSpans` attributes but that was not the case.

**What did you see instead?**
I observed that the `batch.Process` field got mapped to the `ptrace.ResourceSpans` attributes instead of `span.Process`

**What version did you use?**
latest source code

**What config did you use?**
```
receivers:
  jaeger:
    protocols:
      grpc:
        endpoint: "0.0.0.0:14278"

exporters:
  logging:
    loglevel: "debug"

service:
  pipelines:
    traces:
      receivers: [jaeger]
      exporters: [logging]
```

**Environment**
Environment: `Macos/12.2.1` and was also able to reproduce on `Ubuntu 20.04`
Compiler: `go1.17`

**Additional context**
I would like to know what is your take on this problem, I have [opened an issue](https://github.com/open-telemetry/opentelemetry-python/issues/2734) inside the `python-contrib` repository as I though we would like to fill the `batch.Process` field in the `python-contrib` but after further discussions it looked like the `jaeger-receiver` inside the collector should know how to handle a case when `serviceName` field is configured inside `span.Process`

**Comment by jpkrohling** [comment]
I think this is just a simple oversight, but I need more time to look into this.

**Comment by jpkrohling** [comment]
I believe this got fixed by #12297. If this is not the case, let me know and I'll reopen e re-evaluate this.
